src/ray/raylet/worker_pool.cc

```diff
@@ -38,6 +38,28 @@ bool RemoveWorker(std::unordered_set<std::shared_ptr<ray::raylet::Worker>> &work
   return worker_pool.erase(worker) > 0;
 }
 
+// A helper function to remove a worker client from a map of pid to list of clients.
+// If the list of clients is empty after removing this one, erases it from the map.
+bool RemoveConnection(
+    std::unordered_map<pid_t, std::vector<std::shared_ptr<ray::LocalClientConnection>>>
+        pid_to_connections,
+    const pid_t pid, const std::shared_ptr<ray::LocalClientConnection> &connection) {
+  bool found = false;
+  auto it = pid_to_connections.find(pid);
+  if (it != pid_to_connections.end()) {
+    for (size_t i = 0; i < it->second.size(); i++) {
```

Comment:
There is probably a std::remove method or similar to do this with less code, but again no big deal.